### PR TITLE
Csec api with client info case 1169

### PR DESCRIPTION
**Pull request by abelayt**

### CODE REVIEW

**Comment by SashaKyz** [comment]
Can one of the admins verify this patch?

File: app/controllers/api/child_clients_controller.rb [hunk 1]
```diff
@@ -6,5 +6,11 @@ def show
       childclient = ChildClients::ChildClientRepository.new.show(params[:id], session[:token])
       render json: childclient
     end
+
+    def childclients_by_csec
```
Comment (amanbop) [inline app/controllers/api/child_clients_controller.rb]: Just a minor suggestion... we can use underscore like child_clients.. for better readability.

File: app/javascript/Clients/clientIdPage/ClientInformation.js [hunk 2]
```diff
@@ -40,13 +46,14 @@ export default class ClientInformation extends React.Component {
     super(props);
     this.state = {
       response: { XHRStatus: 'idle' },
+      csecresponse: { XHRStatus: 'idle' },
```
Comment (amanbop) [inline app/javascript/Clients/clientIdPage/ClientInformation.js]: Same here.. underscore will improve readability

File: app/javascript/Clients/clientIdPage/ClientInformation.js [hunk 3]
```diff
@@ -64,39 +71,56 @@ export default class ClientInformation extends React.Component {
...
+          nameType: String(response.name_type),
+        })
+      )
+      .catch(() => this.setState({ response: { XHRStatus: 'error' } }));
+  };
+  setCsecData = () => {
+    return ChildClientService.csec()
+      .then(csecresponse =>
+        this.setState({
+          csecresponse,
```
Comment (Sushmareddy-OSI) [inline app/javascript/Clients/clientIdPage/ClientInformation.js]: small change, just make sure to have camelCase to have consistency.

File: app/javascript/Clients/clientIdPage/ClientInformation.js (hunk 2 above)
Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/ClientInformation.js]: I think it should camelCase, not an underscore

Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/ClientInformation.js]: I think It should be camelCase.

**Comment by RamuKammagani-osi** [comment]
Some refactor is need in Routes but we can refactor in the follow up story!

File: config/routes.rb [hunk 4]
```diff
@@ -17,6 +17,8 @@
     resources :child_clients, only: [ :show ] do
     end
 
+    get 'child_clients/:id/csec' => 'child_clients#child_clients_by_csec'
```
Comment (RamuKammagani-osi) [inline config/routes.rb]: This should be under the resources child_clients as member!

Comment (abelayt) [reply config/routes.rb]: Did not we just talked about  this and agreed to keep it? It works!

Comment (RamuKammagani-osi) [reply config/routes.rb]: It works fine! @abelayt Just to be consistent might be refactoring is better! if you wanna refactor in follow up story that is fine too!

Comment (abelayt) [reply config/routes.rb]: @amanbop was part of our discussion with you on this.  Also, I discussed this last week with him this way with 'get', or as a  'member within collection/resources' and we agreed to keep it as is now with 'get'. For me it was the most readable one. If he has changed his mind now at the last hour and wants to use member instead, sure I'll change it.

Comment (RamuKammagani-osi) [reply config/routes.rb]: Sounds good! Thanks for the clarification @abelayt

**Pull request merged**